packages/segmented-control/src/SegmentedControl/SegmentedControl.tsx

```diff
@@ -114,6 +114,12 @@ export const SegmentedControl = forwardRef<
     };
   }, [handleFocusIn]);
 
+  useEffect(() => {
+    if (isControlled) {
+      setInternalValue(controlledValue);
+    }
+  }, [controlledValue, isControlled]);
+
```

Comment:
Can we add a test to make sure this regression doesn't happen again?

Reply:
Done, re-r?

Reply:
c/p this example in storybook as well: 
https://codesandbox.io/s/fragrant-wood-ztxz37?file=/src/App.tsx

Reply:
I don't think these tests are testing the broken behavior. I copied the tests to main and they were still passing

Reply:
I think you'll need to reproduce the container component that's in the sandbox (i.e. with the `useState` hook) and then add the following at the end of the test:
```ts
      // Click the Banana button
      userEvent.click(banana!);
      expect(banana).toHaveAttribute('aria-selected', 'true');
```